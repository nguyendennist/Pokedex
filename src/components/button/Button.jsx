import React from 'react';
import PropTypes from 'prop-types';

const Button = ({name, onClick}) => (
	<input type='button' value={name} onClick={onClick} />
);

Button.propTypes = {
	name: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired
};

export default Button;
