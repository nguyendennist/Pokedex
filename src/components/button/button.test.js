import React from 'react';
import { create } from 'react-test-renderer';
import Button from './Button.jsx';

describe('unit tests', () => {
	test('should be defined', () => {
		expect(Button).toBeDefined();
	});
});

describe('ui tests', () => {
	test('should render correctly', () => {
		const tree = create(
			<Button name='test' onClick={() => {}}/>
		).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
