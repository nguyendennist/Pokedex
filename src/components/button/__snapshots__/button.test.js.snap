// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui tests should render correctly 1`] = `
<input
  onClick={[Function]}
  type="button"
  value="test"
/>
`;
